components/sensor/hmc5883l.rst

```diff
@@ -43,10 +46,63 @@ Configuration variables:
   :ref:`Sensor <config-sensor>`.
 - **heading** (*Optional*): The heading of the sensor in degrees. All options from
   :ref:`Sensor <config-sensor>`.
+- **oversampling** (*Optional*): The oversampling parameter for the sensor.
+- **data_rate** (*Optional*): The data rate parameter for the sensor.
+- **measurement_mode** (*Optional*): The measurement mode parameter for the sensor.
+- **range** (*Optional*): The range parameter for the sensor.
 - **update_interval** (*Optional*, :ref:`config-time`): The interval to check the sensor. Defaults to ``60s``.
-
 - **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
 
+Oversampling Options
+--------------------
+
+By default, the HMC5883L sensor measures each value 1 times when requesting a new value. You can, however,
+configure this amount. Possible sampling values:
+
+-  ``1x`` (default)
+-  ``2x``
+-  ``4x``
+-  ``8x``
+
+Data Rate Options
+-----------------
+
+By default, the HMC5883L sensor measures 15 sampled ever second. You can, however,
+configure this amount. Possible sampling values:
```

Comment:
```suggestion
configure this amount. Possible data rate values:
```

We should not expose this directly to the user, we can calculate the best setting here ourselves by looking at the update_interval value.